src/app/modules/item/pages/item-edit/item-edit.component.ts

```diff
@@ -79,22 +78,8 @@ export class ItemEditComponent implements OnDestroy {
     });
   }
 
-  updateForm(children: ChildData[]): void {
-    this.itemForm.patchValue({
-      children: children.map((child, idx) => ({ item_id: child.id, order: idx })),
-    });
-  }
-
-  getItemChanges(): ItemChanges | undefined {
-    const children = this.itemForm.get('children');
-
-    if (children === null) return undefined;
-
-    const res: ItemChanges = {};
-    const childrenValue = children.value as ItemChild[];
-    if (childrenValue.length) res.children = childrenValue;
-
-    return res;
+  updateItemChanges(children: ChildData[]): void {
+    if (children.length) this.itemChanges.children = children.map((child, idx) => ({ item_id: child.id, order: idx }));
```

Comment:
Why `if (children.length) `?  Will not work as soon as you'll allow removing children.

Reply:
Fixed